### PR TITLE
Mercury

### PR DESCRIPTION
**Pull request by danmanj**

### CODE REVIEW

File: protocol_driver_mercury.cc [hunk 1]
```diff
@@ -36,7 +36,7 @@ ProtocolDriverMercury::ProtocolDriverMercury() {}
 absl::Status ProtocolDriverMercury::Initialize(
     const ProtocolDriverOptions& pd_opts, int* port) {
   std::string netdev_name = pd_opts.netdev_name();
-  auto maybe_ip = IpAddressForDevice(netdev_name, 4);
+  auto maybe_ip = IpAddressForDevice(netdev_name, mercury_ipv4_only_ ? 4 : 0);
```
Comment (kanthicn1) [inline protocol_driver_mercury.cc:39]: Not sure I understand this. What is `mercury_ipv4_only_` ?

File: protocol_driver_mercury.h [hunk 2]
```diff
@@ -78,6 +78,8 @@ class ProtocolDriverMercury : public ProtocolDriver {
 
   hg_context_t* hg_context_ = nullptr;
   hg_class_t* hg_class_ = nullptr;
+  static constexpr bool mercury_ipv4_only_ =
+      (HG_VERSION_MAJOR < 2) || (HG_VERSION_MAJOR == 2 && HG_VERSION_MINOR < 2);
```
Comment (kanthicn1) [inline protocol_driver_mercury.h:82]: This makes sense now. HG_VERSION_MAJOR /MINOR determines this bool which determines the netdev_name above. Is that correct?

File: setup_mercury.sh [hunk 3]
```diff
@@ -1,27 +1,114 @@
...
-wget https://github.com/mercury-hpc/mercury/releases/download/v2.0.1/mercury-2.0.1.tar.bz2 -O mercury-2.0.1.tar.bz2
-tar xvfj mercury-2.0.1.tar.bz2
-cd mercury-2.0.1
-mkdir -p build
-cd build
-cmake DCMAKE_-BUILD_TYPE=Debug -DNA_USE_SM=OFF -DNA_USE_OFI=ON -DOFI_INCLUDE_DIR=$DST_LIBFABRIC/include -DOFI_LIBRARY=$DST_LIBFABRIC/lib -DCMAKE_INSTALL_PREFIX=$DST_MERCURY ..
-make -j 6
-make install
-cd ../../distbench
+set -uEeo pipefail
```
Comment (kanthicn1) [inline setup_mercury.sh:2]: Dan, Is there some place where you have written what mercury driver is and why it is needed? If not, would adding a sentence here make sense?

**Review by kanthicn1 — APPROVED** [review]
Approving since most of it makes sense. I left a  few comments for my own understanding of what the scripts are doing.

File: protocol_driver_mercury.h (hunk 2 above)
Comment (danmanj) [reply protocol_driver_mercury.h:82]: It determines if the protocol driver asks for IP version 4, or allows for any IP version to be used.

File: setup_mercury.sh (hunk 3 above)
Comment (danmanj) [reply setup_mercury.sh:2]: Sure, I'll add the URL to the mercury project page.

File: protocol_driver_mercury.cc (hunk 1 above)
Comment (danmanj) [reply protocol_driver_mercury.cc:39]: your explanation below was correct.

**Pull request merged**